### PR TITLE
Fix end of epoch loss logging bug 

### PR DESCRIPTION
**Pull request by AnnaKwa**
Fixes bug where the training loop callback logged the losses at each timesteps batch step instead of just the end of the epoch.

### CODE REVIEW

**Review by mcgibbon — APPROVED** [review]
Thanks for the unit test!
